test/GcpDLPRedactor.test.ts

```diff
@@ -0,0 +1,170 @@
+import { GoogleDLPRedactor, defaultInfoTypes } from '../src/custom/GoogleDLPRedactor';
+
+function mockDlpProject(dlpRedactor: GoogleDLPRedactor) {
+  Object.assign(dlpRedactor.dlpClient, {
+    getProjectId: () => Promise.resolve('mock-project'),
+    projectPath: () => 'projects/mock-project'
+  });
+}
+
+describe(GoogleDLPRedactor.name, function() {
+  const dlpRedactor = new GoogleDLPRedactor();
+
+  mockDlpProject(dlpRedactor);
+
+  it('can do basic redaction', async function() {
+    const original = "Hey it's David Johnson with ACME Corp. My SSN is 123-45-6789.";
+    const expected = "Hey it's PERSON_NAME with ACME Corp. My SSN is US_SOCIAL_SECURITY_NUMBER.";
+
+    dlpRedactor.dlpClient.inspectContent = () =>
+      Promise.resolve([
+        {
+          result: {
+            findings: [
+              { quote: 'David Johnson', infoType: { name: 'PERSON_NAME' } },
+              { quote: 'David', infoType: { name: 'FIRST_NAME' } },
+              { quote: 'Johnson', infoType: { name: 'LAST_NAME' } },
+              { quote: 'David Johnson', infoType: { name: 'MALE_NAME' } },
+              { quote: '123-45-6789', infoType: { name: 'US_SOCIAL_SECURITY_NUMBER' } }
+            ]
+          }
+        }
+      ]);
+
+    await expect(dlpRedactor.redactAsync(original)).resolves.toBe(expected);
+  });
+
+  it('should not treat input as regex', async function() {
+    const original = 'Just call (646) 846-1111 or (646) 846-3663 or (646) 846-3663.';
```

Comment:
Looks like these are real phone numbers for a real company. We may want to replace these with fake phone numbers although they are public phone numbers from the company's public website. The same comment applies to the other 646 phone numbers in this test file.

Reply:
you're right - how did you catch that? :)